src/passes/Print.cpp

```diff
@@ -2666,7 +2651,7 @@ struct PrintSExpression : public UnifiedExpressionVisitor<PrintSExpression> {
     doIndent(o, indent);
     o << '(';
     printMedium(o, "elem");
-    if (curr->hasExplicitName) {
+    if (curr->hasExplicitName || currModule->elementSegments.size() > 1) {
```

Comment:
How about a comment like:
```suggestion
    // If there is no explicit name, and there are multiple segments, use our
    // internal names to differentiate them.
    if (curr->hasExplicitName || currModule->elementSegments.size() > 1) {
```

Reply:
Done, thanks!